src/Components/EnvironmentPicker/EnvironmentPicker.tsx

```diff
@@ -122,68 +122,59 @@ const EnvironmentPicker = (props: EnvironmentPickerProps) => {
             type: DialogType.normal,
             title: t('environmentPicker.editEnvironment'),
             closeButtonAriaLabel: t('close'),
-            subText: props.isLocalStorageEnabled
-                ? (props.storage
+            subText: isLocalStorageEnabled
+                ? (storage
                       ? t('environmentPicker.descriptionForEnvAndCont')
                       : t('environmentPicker.descriptionForEnvironment')) +
                   ' ' +
                   t('environmentPicker.descriptionForLocalStorage')
-                : props.storage
+                : storage
                 ? t('environmentPicker.descriptionForEnvAndCont')
                 : t('environmentPicker.descriptionForEnvironment')
         }),
-        [t]
+        [t, isLocalStorageEnabled, storage]
     );
 
-    const previousIsDialogHidden = usePrevious(props.isDialogHidden);
+    const previousIsDialogHidden = usePrevious(isDialogHidden);
     // Figure out if dialog needs to be open from props
     useEffect(() => {
         // Have undefined checked onMount to avoid an extra render
         // Have is previous check from true to false, to just change dialogHidden on open
         if (
             previousIsDialogHidden !== undefined &&
             previousIsDialogHidden === true &&
-            previousIsDialogHidden !== props.isDialogHidden
+            previousIsDialogHidden !== isDialogHiddenProp
```

Comment:
hmm, this piece of code was from earlier time to prevent rerendering by Pato, now I removed the dialoghidden prop from EnvironmenPicker since I think it should be only handled internally by a state variable now (keep in mind that EnvironmentPicker component has two pieces: 1- is the header environment texts with edit icon to open the dialog, 2- the dialog with ResourcePicker components). Does it make sense? Changes: https://github.com/microsoft/iot-cardboard-js/pull/684/commits/7fd8facb79e2d0e8bdb1507b7ff64b355140bc07

FYI @pasanchMSFT I removed your code block to trigger rerender based on a dialoghidden prop change since now it is only handled internally by the EnvironmentPicker component: https://github.com/microsoft/iot-cardboard-js/pull/684/commits/7fd8facb79e2d0e8bdb1507b7ff64b355140bc07#diff-2e58d2278063e505656fe7c499cb7bb0769943ccd89ac6f49417ac4be03cbea0L125